substrate/frame/offences/src/lib.rs

```diff
@@ -73,14 +73,28 @@ pub mod pallet {
 
 	/// The primary structure that holds all offence records keyed by report identifiers.
 	#[pallet::storage]
-	#[pallet::getter(fn reports)]
 	pub type Reports<T: Config> = StorageMap<
 		_,
 		Twox64Concat,
 		ReportIdOf<T>,
 		OffenceDetails<T::AccountId, T::IdentificationTuple>,
 	>;
 
+	impl<T: Config> Pallet<T> {
+		#[deprecated(note = "Use Reports::<T>::get(id) instead.")]
```

Comment:
but there is no need to deprecate here.

Just keep this public function around forever. The point is just to remove the macro logic, not necessarily remove this functionality.

Reply:
Ah that was my request - that was the impression that I got from the discussion on the issue, maybe I'm not up to date on the intention behind removing the getters